libsnark/zk_proof_systems/plonk/prover.hpp

```diff
@@ -0,0 +1,252 @@
+/** @file
+*****************************************************************************
+Declaration of Prover interfaces for ppzkSNARK proof system Plonk.
+
+This includes:
+- class for proof
+- class for prover
+
+References:
+
+\[GWC19]:
+"Plonk: Permutations over lagrange-bases for oecumenical noninteractive
+arguments of knowledge", Ariel Gabizon, Zachary J. Williamson, and Oana
+Ciobotaru, Cryptology ePrint Archive, Report 2019/953, 2019,
+<https://eprint.iacr.org/2019/953>
+
+*****************************************************************************
+* @author     This file is part of libsnark, developed by SCIPR Lab
+*             and contributors (see AUTHORS).
+* @copyright  MIT license (see LICENSE file)
+*****************************************************************************/
+
+#ifndef PLONK_PPZKSNARK_PROVER_HPP_
+#define PLONK_PPZKSNARK_PROVER_HPP_
+
+namespace libsnark
+{
+/**
+ * A proof for the Plonk GG-ppzkSNARK.
+ */
```

Comment:
Let's make sure that we use `///` for documentation comments (same applies for other files in the PR).

Reply:
Fixed in all files